static/app/components/devtoolbar/hooks/useReplayRecorder.tsx

```diff
@@ -0,0 +1,117 @@
+import {useCallback, useEffect, useState} from 'react';
+import type {replayIntegration} from '@sentry/react';
+import type {ReplayRecordingMode} from '@sentry/types';
+
+import useConfiguration from 'sentry/components/devtoolbar/hooks/useConfiguration';
+import {useSessionStorage} from 'sentry/utils/useSessionStorage';
+
+type ReplayRecorderState = {
+  disabledReason: string | undefined;
+  isDisabled: boolean;
+  isRecording: boolean;
+  lastReplayId: string | undefined;
+  start(): Promise<boolean>;
+  stop(): Promise<boolean>;
+};
+
+interface ReplayInternalAPI {
+  [other: string]: any;
+  getSessionId(): string | undefined;
+  isEnabled(): boolean;
+  recordingMode: ReplayRecordingMode;
+}
+
+function getReplayInternal(
+  replay: ReturnType<typeof replayIntegration>
+): ReplayInternalAPI {
+  // While the toolbar is internal, we can use the private API for added functionality and reduced dependence on SDK release versions
+  // @ts-ignore:next-line
+  return replay._replay;
+}
+
+const LAST_REPLAY_STORAGE_KEY = 'devtoolbar.last_replay_id';
+
+export default function useReplayRecorder(): ReplayRecorderState {
+  const {SentrySDK} = useConfiguration();
+  const replay =
+    SentrySDK && 'getReplay' in SentrySDK ? SentrySDK.getReplay() : undefined;
```

Comment:
you can use `SentrySDK.getIntegrationByName()`, it's just a bit more annoying dealing with types

Reply:
Hm my VSCode can't find this method in browser or react packages..

Reply:
Got it, does this have any improvements over the current approach? Right now I'm using these conditions to get the `disabledReason` (see screenshots I just added in the description)

Reply:
Ah just as a fallback if the SDK is too old to support `getReplay` as it was a relatively recent addition, but it's fine not to support older